### PR TITLE
roles:  adding optional email parameter to active directory user add

### PR DESCRIPTION
**Pull request by danlavu**

### CODE REVIEW

**Comment by danlavu** [comment]
We have a test case worth automating https://github.com/SSSD/sssd/pull/7174 , using the email instead of the userid to login. To adequately test this, we need to be able to change the email value of the user created in AD.

**Comment by danlavu** [comment]
Test case https://github.com/SSSD/sssd/pull/7206

**Comment by pbrezina** [comment]
LGTM, can you please add it also to Samba? There's `--mail-address` parameter. It would also be nice if we are able to change it in `modify()`.

File: sssd_test_framework/roles/ad.py [hunk 1]
```diff
@@ -610,9 +611,14 @@ def add(
         :type gecos: str | None, optional
         :param shell: Login shell, defaults to None
         :type shell: str | None, optional
+        :param email: Email, defaults to None
```
Comment (pbrezina) [inline sssd_test_framework/roles/ad.py:614]: `defaults to None (= user@domain)`, same in samba

**Review by pbrezina — CHANGES_REQUESTED** [review]

**Pull request merged**